clang/lib/Driver/ToolChains/Clang.cpp

```diff
@@ -1212,6 +1215,22 @@ void Clang::AddPreprocessingOptions(Compilation &C, const JobAction &JA,
       CmdArgs.push_back("-module-file-deps");
   }
 
+  if (!ArgM && Args.hasArg(options::OPT_fintelfpga) &&
+      JA.isDeviceOffloading(Action::OFK_SYCL)) {
+    // No dep generation option was provided, add all of the needed options
+    // to ensure a successful dep generation.
+    const char *DepFile;
+    createFPGATempDepFile(DepFile);
+    CmdArgs.push_back("-dependency-file");
+    CmdArgs.push_back(DepFile);
+    CmdArgs.push_back("-MT");
+    SmallString<128> P(Inputs[0].getBaseInput());
+    llvm::sys::path::replace_extension(P, "o");
+    SmallString<128> Quoted;
+    QuoteTarget(llvm::sys::path::filename(P), Quoted);
+    CmdArgs.push_back(Args.MakeArgString(Quoted));
```

Comment:
Can we double-check these in the test?

Reply:
I'll adjust the test accordingly.  As for your `-foffload-static-lib` issue with `.d` question, this patch does not address that problem.  For that problem, we need to extract the .d files from the archive and add them to the aoc call.  Current device extraction only takes into account the device objects and passes them along to the partial link step.  The `.d` files are left alone.

Reply:
I suppose this also affect offload static library that are passed without this flag?

Reply:
@meiyacha, that is correct.  Offload static lib w/ and w/out -foffload-static-lib behaves the same.